aiohttp/_websocket.pyx

```diff
@@ -1,20 +1,51 @@
-from cpython cimport PyBytes_FromStringAndSize, PyBytes_AsString
-from cpython.ref cimport PyObject
+from cpython cimport PyBytes_AsString
+import platform
 
+#from cpython cimport PyByteArray_AsString # cython still not exports that
 cdef extern from "Python.h":
-    char* PyByteArray_AsString(object bytearray) except NULL
+    char* PyByteArray_AsString(bytearray ba) except NULL
 
+from libc.stdint cimport uint32_t, uint64_t
 
 def _websocket_mask_cython(bytes mask, bytearray data):
-    cdef Py_ssize_t mask_len, data_len, i
-    cdef char * in_buf
-    cdef char * out_buf
-    cdef char * mask_buf
-    cdef bytes ret
-    mask_len = len(mask)
+    """Note, this function mutates it's `data` argument
+    """
+    cdef:
+        Py_ssize_t data_len, i
+        # bit operations on signed integers are implementation-specific
+        unsigned char * in_buf
+        const unsigned char * mask_buf
+        uint32_t uint32_msk;
+        uint64_t uint64_msk;
+
+    assert len(mask) == 4
+
+
     data_len = len(data)
-    in_buf = PyByteArray_AsString(data)
-    mask_buf = PyBytes_AsString(mask)
+    in_buf = <unsigned char*>PyByteArray_AsString(data)
+    mask_buf = <const unsigned char*>PyBytes_AsString(mask)
+    uint32_msk = (<uint32_t*>mask_buf)[0]
+
+    # TODO: align in_data ptr to acheive even faster speeds
+    # does it need in python ?! malloc always aligns to sizeof(long) bytes
+
+    (bits, linkage) = platform.architechure
+    if bits == '64bit':
```

Comment:
fixed as sizeof(uintmax_t). on some arches pointers are 32-bit, while native arithmetics is 64-bit...